dask_planner/src/parser.rs

```diff
@@ -1174,14 +1182,23 @@ impl<'a> DaskParser<'a> {
     /// Parse Dask-SQL SHOW COLUMNS FROM <table>
     fn parse_show_columns(&mut self) -> Result<DaskStatement, ParserError> {
         self.parser.expect_keyword(Keyword::FROM)?;
-        let (tbl_schema, tbl_name) =
+        let (schema_name, table_name) =
             DaskParserUtils::elements_from_object_name(&self.parser.parse_object_name()?)?;
         Ok(DaskStatement::ShowColumns(Box::new(ShowColumns {
-            table_name: tbl_name,
-            schema_name: match tbl_schema.as_str() {
-                "" => None,
-                _ => Some(tbl_schema),
-            },
+            schema_name,
+            table_name,
+        })))
+    }
+
+    /// Parse Dask-SQL SHOW MODEL [FROM <schema>]
+    fn parse_show_models(&mut self) -> Result<DaskStatement, ParserError> {
+        let mut schema_name: Option<String> = None;
+        if !self.parser.consume_token(&Token::EOF) {
+            self.parser.expect_keyword(Keyword::FROM)?;
+            schema_name = Some(self.parser.parse_identifier()?.value);
```

Comment:
nit: it might be good to add a check for an expected EOF after the schema name

Reply:
I guess this comment applies to all cases where we are taking over parsing. If we don't check for EOF after parsing then we are potentially silently ignoring any extra tokens.

Reply:
Yeah that's a good point - considering the diff of this PR is already pretty large, are you okay with me opening an issue to track this and we can address it in a follow up?

Reply:
Yes, absolutely.

Reply:
It looks like maybe a check for the end of statement is being done implicitly? Trying out the following with this branch:

```python
from dask_sql import Context

c = Context()

c.sql("show models from 'root' extra tokens")
```

I get a `ParserError("Expected end of statement, found: extra")`.